Drop commontorizon from Toradex supported

microhobby opened the pull request:
From feedback is not good to have experimental stuff or release Toradex official stuff “crossover-ed” with experimental stuff. So, we need to drop everything experimental from the Toradex supported templates.

@andreriesco this will be added only on the TOS 6, so this will include also what is not in sync yet from `dev`. This should be the last release on `bookworm` and will enter in maintenance mode.

microhobby committed:
cmakeConsole: Drop commontorizon of the Toradex supported

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cppConsole: Drop commontorizon from Toradex supported

side effect, drop x86_64 support

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
cppQML: Drop commontorizon from Toradex supported

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetConsole: Drop commontorizon from Toradex supported

Side effect drop the x86_64 support from the project

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
aspnetBlazor: Drop commontorizon from Toradex supported

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
python3Console: Drop commontorizon from Toradex supported

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
genericTemplate: Drop commontorizon from Toradex supported

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

andreriesco commented on cppConsole/.vscode/launch.json:
```diff
@@ -26,50 +26,6 @@
-            "preLaunchTask": "deploy-torizon-riscv64"
-        },
         {
             "name": "Torizon AMD64",
```
Also this one can be removed

microhobby replied:
Done in 3763f80171065167be1e385a8aa90dadc92b88f4

andreriesco replied:
Just for `cConsole` on this commit, for `cppConsole` it still needs to be removed

microhobby replied:
Check it again, there is no `deploy-torizon-riscv64` for `cppConsole`

andreriesco replied:
Not `riscv64`, the `Torizon AMD64` one

microhobby replied:
ok, so, `risc-v` was removed from everywhere. `x86_64` only from Toradex supported.

microhobby committed:
all: Remove riscv64 support from the templates

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: installDepsScripts: installDotnetSDK8: Fix package not found

If the package is not there the apt will fail, and we are enforcing the
exit on error, so we need to disable it before try to check if there
is a package to remove.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

andreriesco commented on README.md:
```diff
@@ -6,17 +6,17 @@ This repository maintains the templates used in conjunction with the [VS Code To
 | TEMPLATE | DESCRIPTION | RUNTIME | LANGUAGE | HW ARCH | FOLDER |
 | -------- | ----------- | ------- | -------- | ------- | ------ |
-| ![](https://raw.githubusercontent.com/toradex/vscode-torizon-templates-documentation/refs/heads/main/thumbnails/cppconsole.png?raw=true) | C++ Makefile Console Application | libc | C++ | ![](assets/img/arm32.png?raw=true&id=2) ![](assets/img/arm64.png?raw=true&id=2) ![](assets/img/riscv64.png?raw=true&id=2) ![](assets/img/x8664.png?raw=true&id=2) | [cppConsole](./cppConsole) |
+| ![](https://raw.githubusercontent.com/toradex/vscode-torizon-templates-documentation/refs/heads/main/thumbnails/cppconsole.png?raw=true) | C++ Makefile Console Application | libc | C++ | ![](assets/img/arm32.png?raw=true&id=2) ![](assets/img/arm64.png?raw=true&id=2) ![](assets/img/x8664.png?raw=true&id=2) | [cppConsole](./cppConsole) |
```
Also the `![](assets/img/x8664.png?raw=true&id=2)` needs to be removed from the templates where it has been removed.

microhobby replied:
This should be fixed on 7258c62

andreriesco commented on scripts/installDepsScripts/installDotnetSDK8.sh:
```diff
@@ -50,9 +50,10 @@ if [ "$source" != "https://packages.microsoft.com/$repo/$repo_version/prod" ]; t
     rm packages-microsoft-prod.deb
 
     # Remove the dotnet-sdk installation that doesn't come from the Microsoft source
+    set +e
```
Does it makes sense to add also this fix on `installDotnetSDK7` script, which is used by Avalonia templates?

microhobby replied:
yes make sense, added here 3313e68

andreriesco commented on cppConsole/Dockerfile:
```diff
@@ -12,7 +12,7 @@ ARG APP_ROOT=
 
 # BUILD ------------------------------------------------------------------------
-FROM commontorizon/debian-cross-toolchain-${IMAGE_ARCH}:${CROSS_SDK_BASE_TAG} AS build
+FROM torizon/debian-cross-toolchain-${IMAGE_ARCH}:${CROSS_SDK_BASE_TAG} AS build
```
Here we should use `commontorizon` as an exception, because the container for arm is `debian-cross-toolchain-armhf` instead of `debian-cross-toolchain-arm`. Also, the tags are different for `debian-cross-toolchain-armhf` and `debian-cross-toolchain-arm64`. `3.2.0` for example is not available for `debian-cross-toolchain-armhf, `3.3.0` and beyond is not available for `debian-cross-toolchain-arm64`, one option for this problem would be to use just the major `3` though.

microhobby replied:
This should be handled by 86d22b3

microhobby replied:
Some adjustment on build steps need to be also added: 6018428

andreriesco replied:
Sounds good this adjustments

microhobby committed:
tasks: common: Drop commontorizon/binfmt usage

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tasks: common: Use armhf as IMAGE_ARCH for 32bit arm

This is part of the drop of commontorizon images.
The Toradex images have the prefix armhf for the 32bit arm architecture
and the linux/armhf also works as a valid architecture for the docker
platform.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
README: Remove x86_64 from the templates list

The x86_64 architecture is not supported by Toradex images, and
the experiments here are over. So, we should remove the x86_64
architecture from the templates list.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: installDepsScripts: installDotnetSDK7: Fix package not found

If the package is not there the apt will fail, and we are enforcing the
exit on error, so we need to disable it before try to check if there
is a package to remove.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: Dockerfile: Adjust the build steps to use armhf instead arm

The Toradex cross toolchain images use the prefix armhf instead of arm,
and the templates are expecting arm as the prefix for the armv7 32bit.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnet: Dockerfile: Make the move of the publish folder generic

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tasks: common: Add CROSS_SDK_BASE_TAG to the build args

As there is no consistency in the Toradex official images, we need to
specify the CROSS_SDK_BASE_TAG to the build args to ensure that the
correct version of the SDK is used.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: createDockerComposeProduction: Fix arm 32bit architecture

Following the changes for have the arm 32bit architecture as armhf
instead of arm, this commit updates the script to reflect this change.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: createDockerComposeProduction: Use tasks to build release

This also need a change on the docker_tag setting. This is now
pre-filled with v0.0.0, and then the user need to change it to the
desired tag.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnet*: Dockerfile: Fix edge case when ARCH and IMAGE_ARCH are equal

Like on arm64, the ARCH and IMAGE_ARCH are equal, so the mv command
fails, this patch adds a if to check if the paths are different before
moving the files.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnet*: conf: Add Dockerfile to update.json

With the last changes on how the arch is handled on the tasks, the
Dockerfile for production needs update.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
aspnetBlazor: Dockerfile: Fix edge case when ARCH and IMAGE_ARCH are =

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
templates: remove Torizon AMD64 debugger from templates

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>
